core/src/main/java/com/novoda/merlin/service/MerlinService.java

```diff
@@ -30,7 +30,17 @@ public IBinder onBind(Intent intent) {
     @Override
     public boolean onUnbind(Intent intent) {
         isBound = false;
-        connectivityChangesRegister.unregister();
+        if (connectivityChangesRegister != null) {
```

Comment:
I'd prefer to get rid of these checks if I am completely honest, is this likely to occur @rock3r that you know of?